model_compression_toolkit/target_platform_capabilities/schema/v1.py

```diff
@@ -499,40 +666,55 @@ class TargetPlatformModel:
         tpc_platform_type (Optional[str]): Type of the platform for the Target Platform Configuration.
         add_metadata (bool): Flag to determine if metadata should be added.
         name (str): Name of the Target Platform Model.
-        operator_set (Tuple[OperatorsSetBase]): Tuple of operator sets within the model.
-        fusing_patterns (Tuple[Fusing]): Tuple of fusing patterns for the model.
+        operator_set (Tuple[OperatorsSetBase, ...]): Tuple of operator sets within the model.
+        fusing_patterns (Tuple[Fusing, ...]): Tuple of fusing patterns for the model.
         is_simd_padding (bool): Indicates if SIMD padding is applied.
         SCHEMA_VERSION (int): Version of the schema for the Target Platform Model.
     """
     default_qco: QuantizationConfigOptions
-    tpc_minor_version: Optional[int]
-    tpc_patch_version: Optional[int]
-    tpc_platform_type: Optional[str]
+    tpc_minor_version: Optional[int] = None
+    tpc_patch_version: Optional[int] = None
+    tpc_platform_type: Optional[str] = None
     add_metadata: bool = True
-    name: str = "default_tp_model"
-    operator_set: Tuple[OperatorsSetBase] = None
-    fusing_patterns: Tuple[Fusing] = None
+    name: Optional[str] = "default_tp_model"
+    operator_set: Optional[Tuple[OperatorsSet, ...]] = None
+    fusing_patterns: Optional[Tuple[Fusing, ...]] = None
     is_simd_padding: bool = False
 
     SCHEMA_VERSION: int = 1
 
-    def __post_init__(self):
+    model_config = ConfigDict(frozen=True)
+
+    @model_validator(mode='after')
+    def validate_after_initialization(cls, tp_model: 'TargetPlatformModel') -> 'TargetPlatformModel':
         """
-        Post-initialization processing for input validation.
+        Perform validation after the model has been instantiated.
+
+        Args:
+            tp_model (TargetPlatformModel): The instantiated target platform model.
 
-        Raises:
-            Logger critical if the default_qco is not an instance of QuantizationConfigOptions
-            or if it contains more than one quantization configuration.
+        Returns:
+            TargetPlatformModel: The validated model.
         """
         # Validate `default_qco`
-        if not isinstance(self.default_qco, QuantizationConfigOptions):
+        default_qco = tp_model.default_qco
+        if not isinstance(default_qco, QuantizationConfigOptions):
             Logger.critical("'default_qco' must be an instance of QuantizationConfigOptions.") # pragma: no cover
-        if len(self.default_qco.quantization_configurations) != 1:
+
+        if len(default_qco.quantization_configurations) != 1:
```

Comment:
shouldnt it be checked at QuantizationConfigOptions level?

Reply:
this is the default qco so it should have only one option. a regular qco can have multiple options